### PR TITLE
Build LLVM 14 on Yggdrasil

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

**Comment by giordano** [comment]
```
[17:48:55]  ---> ninja -j${nproc} llvm-tblgen clang-tblgen mlir-tblgen mlir-linalg-ods-gen llvm-config
[17:48:56] ninja: unknown target 'mlir-linalg-ods-gen'
```

**Comment by giordano** [comment]
i686-linux-musl is incredibly frustrating: we can't even build Julia itself for that platform :grimacing:

**Comment by vchuravy** [comment]
Oh so I can remove support for that? xD

**Comment by giordano** [comment]
:smiling_face_with_tear:

File: L/LLVM/LLVM_full@14.0.2/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version = v"14.0.2"
+
+include("../common.jl")
+
+
+
+build_tarballs(ARGS, configure_build(ARGS, version; experimental_platforms=true)...;
```
Comment (giordano) [inline L/LLVM/LLVM_full@14.0.2/build_tarballs.jl:7]: `experimental_platforms=true` doesn't do anything special anymore, but it also doesn't hurt

File: L/LLVM/LLVM_full@14.0.2/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+               preferred_gcc_version=v"7", preferred_llvm_version=v"12", julia_compat="1.9")
```
Comment (giordano) [inline L/LLVM/LLVM_full@14.0.2/build_tarballs.jl:8]: For what is worth, now we have also LLVM 13.

**Review by giordano — APPROVED** [review]

**Pull request merged**